src/read_next/read_next.jsx

```diff
@@ -16,29 +19,78 @@ const ReadNext = () => {
                 setPosts({ posts: postsWithoutCurrent });
             });
     }, [params]);
+    const PerPageChange = (value) => {
+        setSize(value);
+        const newPerPage = data.posts ? Math.ceil(data.posts.length / value) : 1;
+        if (current > newPerPage) {
+            setCurrent(newPerPage);
+        }
+    }
+    const getData = (current, pageSize) => {
+        return data.posts ? data.posts.slice((current - 1) * pageSize, current * pageSize) : 1;
+    };
+    const PaginationChange = (page, pageSize) => {
+        setCurrent(page);
+        setSize(pageSize)
+    }
+
+    const PrevNextArrow = (current, type, originalElement) => {
+        if (type === 'prev') {
+            return <button className="pagenation_button_post">Prev</button>;
+        }
+        if (type === 'next') {
+            return <button className="pagenation_button_post">Next</button>;
+        }
+        return originalElement;
+    }
     return (
-        <section className="additional_articles">
-            <h2 className="read_next">What to read next</h2>
-            <div className="photo">
-                {data.posts.map((post) => (
-                    <Link to={"/posts/" + post.id} className="decoration">
-                        &nbsp;
-                        <div className="atribute">
-                            <img
-                                className="second_page_photos"
-                                width="304"
-                                height="176"
-                                src={post.thumbnail_url}
-                                alt={post.title}
-                            />
-                            <div>
-                                <p className="text_after_images">{post.title}</p>
-                            </div>
-                        </div>
-                    </Link>
-                ))}
-            </div>
-        </section>
+
+        <>
+            <section className="additional_articles">
+                <h2 className="read_next">What to read next</h2>
+                <div className="photo">
+                    {
+                        getData(current, size).map((post, index) => {
+                            return (
+                                <Link onClick={window.scrollTo({
+                                    top: 0,
+                                    left: 0,
+                                    behavior: 'smooth',
+                                })} to={"/posts/" + post.id} className="decoration">
```

Comment:
scroll to top should be implemented for the whole application, not just here. you can take a look at the example [here](https://stackoverflow.com/questions/36904185/react-router-scroll-to-top-on-every-transition)